Modules/Scripted/Endoscopy/Endoscopy.py

```diff
@@ -984,6 +908,53 @@ def buildCameraMatrix4x4(cameraPosition, inputMatrix3x3):
             outputMatrix4x4.SetElement(col, 3, cameraPosition[col])
         return outputMatrix4x4
 
+    @staticmethod
+    def getCameraOrientationsFromInputCurve(inputCurve):
+        cameraOrientationsString = inputCurve.GetAttribute(
+            EndoscopyLogic.NODE_PATH_CAMERA_ORIENTATIONS_ATTRIBUTE_NAME,
+        )
+        convertedCameraOrientations = (
+            dict() if cameraOrientationsString is None else json.loads(cameraOrientationsString)
+        )
+        if not EndoscopyLogic.cameraOrientationsIsCorrectType(convertedCameraOrientations):
+            raise ValueError(f"cameraOrientations for curve {inputCurve.GetID()} is not of the correct type.")
+        # JSON turns the `keys` into strings and cannot handle `values` that are numpy arrays.
+        # So, let's convert back.
+        cameraOrientations = {float(k): np.array(v) for k, v in convertedCameraOrientations.items()}
+
+        return cameraOrientations
+
+    @staticmethod
+    def setInputCurveCameraOrientations(inputCurve, cameraOrientations):
+
+        # Rather than a length-zero or length-two string representing an empty dictionary, clear the string.
+        if not cameraOrientations:
+            cameraOrientationsString = None
+
+        # JSON turns float `keys` into strings and cannot handle `values` that are numpy arrays.
+        # So, let's convert.
+        convertedCameraOrientations = {str(k): list(v) for k, v in cameraOrientations.items()}
+        if not EndoscopyLogic.cameraOrientationsIsCorrectType(convertedCameraOrientations):
+            raise ValueError("cameraOrientations is not of the correct type")
+        cameraOrientationsString = json.dumps(convertedCameraOrientations)
```

Comment:
These lines should be in an `else:` block associated with the preceding `if not cameraOrientations:`.